testutil/timeouts.go

```diff
@@ -31,12 +32,15 @@ var runtimeGOARCH = runtime.GOARCH
 //
 // This should only be used in tests and is a bit of a guess.
 func HostScaledTimeout(t time.Duration) time.Duration {
-	switch runtimeGOARCH {
-	case "riscv64":
+	switch {
+	case runtimeGOARCH == "riscv64":
 		// virt riscv64 builders are 5x times slower than
 		// armhf when building golang-1.14. These tests
 		// timeout, hence bump timeouts by 6x
 		return t * 6
+	case os.Getenv("GO_TEST_RACE") == "1":
+		// the -race detector makes test execution time 2-20x slower
+		return t * 5
```

Comment:
Also, if `riscv64` and `-race`, shouldn't the increases in duration be multiplied?

Reply:
it's unlikely someone would run such setup, unless they have lots of patience :smile:

Reply:
I considered complicating this a bit (mentioned in the comment above) but it didn't seem worth it. If a unit test is taking 3min to run we already have bigger problems so I'd say let's just see if this works and adjust as needed